app/src/main/java/com/monzo/androidtest/articles/presentation/ArticlesActivity.kt

```diff
@@ -31,20 +30,23 @@ class ArticlesActivity : AppCompatActivity(), ArticleAdapter.OnItemClickListener
 
         viewModel = HeadlinesApp.from(applicationContext).inject(this)
 
-        adapter = ArticleAdapter(this, this)
+        sectionAdapter = SectionAdapter(this, this)
+
         recyclerView.layoutManager = LinearLayoutManager(this)
-        recyclerView.adapter = adapter
+        recyclerView.adapter = sectionAdapter
 
         swipeRefreshLayout.setOnRefreshListener { viewModel.onRefresh() }
 
         viewModel.state.observe(this) { state ->
             swipeRefreshLayout.isRefreshing = state.refreshing
-            adapter.showArticles(state.articles)
+            sectionAdapter.showSections(state.articleSection)
         }
     }
 
-    override fun onItemClick(position: Int) {
-        val url = viewModel.state.value?.articles?.get(position)?.url
+    override fun onItemClick(articlePosition: Int, sectionPosition: Int) {
+        val url = viewModel.state.value?.articleSection?.get(sectionPosition)?.articles?.get(
+            articlePosition
+        )?.url
```

Comment:
Ai invés de retornar `articlePosition` e `sectionPosition` vamos retornar a própria URL
ou seja:
```suggestion
    override fun onItemClick(articleUrl: String) {
```